nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ValidateCsv.java

```diff
@@ -0,0 +1,613 @@
+                                }
+                            }
+
+                        } catch (final SuperCsvCellProcessorException e) {
+                            valid.set(false);
+                            if(isWholeFFValidation) {
+                                logger.debug("Failed to validate {} against schema due to {}; routing to 'invalid'", new Object[]{flowFile}, e);
+                                break;
+                            } else {
+                                // we append the invalid line to the flow file that will be routed to invalid relationship
+                                invalidFF.set(session.append(invalidFF.get(), new OutputStreamCallback() {
+                                    @Override
+                                    public void process(OutputStream out) throws IOException {
+                                        out.write(print(e.getCsvContext().getRowSource(), csvPref, isFirstLine.get()));
+                                    }
+                                }));
+
+                                if(isFirstLine.get()) {
+                                    isFirstLine.set(false);
+                                }
+                            }
+                        } finally {
+                            if(!isWholeFFValidation) {
+                                totalCount.set(totalCount.get() + 1);
+                            }
+                        }
+                    }
+
+                } catch (final IOException e) {
+                    valid.set(false);
+                    logger.error("Failed to validate {} against schema due to {}", new Object[]{flowFile}, e);
+                } finally {
+                    if(listReader != null) {
+                        listReader.close();
+                    }
+                }
+            }
+        });
+
+        if(isWholeFFValidation) {
+            if (valid.get()) {
+                logger.info("Successfully validated {} against schema; routing to 'valid'", new Object[]{flowFile});
+                session.getProvenanceReporter().route(flowFile, REL_VALID);
+                session.transfer(flowFile, REL_VALID);
+            } else {
+                session.getProvenanceReporter().route(flowFile, REL_INVALID);
+                session.transfer(flowFile, REL_INVALID);
+            }
+        } else {
+            if (valid.get()) {
+                logger.debug("Successfully validated {} against schema; routing to 'valid'", new Object[]{validFF.get()});
+                session.getProvenanceReporter().route(validFF.get(), REL_VALID, "All " + totalCount.get() + " line(s) are valid");
+                session.transfer(validFF.get(), REL_VALID);
+                session.remove(invalidFF.get());
+                session.remove(flowFile);
+            } else if (okCount.get() != 0) {
+                if(header) {
+                    totalCount.set(totalCount.get() - 1);
```

Comment:
This gets decremented here but, I may just be missing it, I don't see the Header add to the totalCount.